dcc-portal-ui/app/scripts/oncogrid/js/controller.js

```diff
@@ -113,4 +113,157 @@
         };
 
     });
+
+    module.controller('OncoGridUploadController', function(Restangular, SetNameService, donorsLimit, genesLimit, filters, 
+        LocationService, $scope, $modalInstance, $location, $q, SetService, $timeout){
+
+        $scope.donorsLimit = donorsLimit;
+        $scope.genesLimit = genesLimit;
+        $scope.filters = filters;
+        $scope.isLaunchingOncoGrid = false;
+
+        $scope.maxDonorsLimit = 3000;
+        $scope.maxGenesLimit = 100;
+
+        $scope.params = {};
+        $scope.hasValidParams= false;
+
+        $scope.params.donorsCount = Math.min($scope.donorsLimit || 3000, $scope.maxDonorsLimit);
+        $scope.params.genesCount = Math.min($scope.genesLimit || 100, $scope.maxGenesLimit);
+        $scope.params.setName = '';
+
+        $scope.hasValidDonorCount = function(value){
+            var val = parseInt(value,10);
+            if (isNaN(val)) {
+                return false;
+            }
+            if (!angular.isNumber(val) || val > $scope.maxDonorsLimit || val <= 0 || val > $scope.donorsLimit) {
+                return false;
+            }
+            return true;
+        }
+
+        $scope.hasValidGeneCount = function(value){
+            var val = parseInt(value,10);
+            if (isNaN(val)) {
+                return false;
+            }
+            if (!angular.isNumber(val) || val > $scope.maxGenesLimit || val <= 0 || val > $scope.genesLimit) {
+                return false;
+            }
+            return true;
+        }
+
+        $scope.checkInput = function() {
+            var params = $scope.params;
+            if ($scope.hasValidDonorCount(params.donorsCount) && 
+                $scope.hasValidGeneCount(params.genesCount)) {
+                $scope.hasValidParams = true;
+            } else {
+                $scope.hasValidParams = false;
+            }
+        };
+
+        $scope.getSetName = function(filters){
+            return SetNameService.getSetFilters()
+                .then(function (filters) {
+                    return SetNameService.getSetName(filters);
+                })
+                .then(function (setName) {
+                    $scope.params.setName = setName;
+                });
+        }
+
+        $scope.getDonorsParams = function(){
+
+            var donorSetParams = {
+                filters: $scope.filters || {},
+                size: $scope.params.donorsCount,
+                type: 'donor',
+                isTransient: true,
+                name: `Top ${$scope.params.donorsCount} Donors: ${_.includes($scope.params.setName, 'All') ? '' : $scope.params.setName}`
+            };
+
+            return donorSetParams;
+        }
+
+        $scope.getGenesParams = function(){
+
+             var geneSetParams = {
```

Comment:
Again, same pattern as above. We should generalize for reuse and reduce duplication / clutter